components/ConnectWalletButton.tsx

```diff
@@ -61,70 +37,23 @@ const ConnectWalletButton = (props) => {
     }
   };
 
-  const { show } = useWalletIdentity();
-
-  const { displayName } = useAddressName(
-    connection.current,
-    current?.publicKey || undefined,
-  );
-
-  const walletAddressFormatted = current?.publicKey
-    ? abbreviateAddress(current?.publicKey)
-    : '';
-
   return (
     <div className="flex">
       <button
-        disabled={connected}
-        className={`bg-transparent border border-fgd-4 border-r-0 default-transition flex h-12 items-center pl-1 pr-2 rounded-l-full rounded-r-none ${
-          connected ? 'cursor-default' : 'hover:bg-bkg-3 focus:outline-none'
-        }`}
+        className={`bg-transparent border border-fgd-3 border-r-0 default-transition flex h-12 items-center pl-3 pr-4 rounded-l-full rounded-r-none w-36 hover:bg-bkg-3 focus:outline-none`}
         onClick={handleConnectDisconnect}
         {...props}
       >
-        <div className="flex font-bold items-center text-fgd-1 text-left text-sm relative">
-          {connected && current?.publicKey ? (
-            <div className="w-12 pr-2">
-              <AddressImage
-                dark={true}
-                connection={connection.current}
-                address={current?.publicKey}
-                height="40px"
-                width="40px"
-                placeholder={
-                  <div className="bg-bkg-4 flex flex-shrink-0 items-center justify-center h-10 rounded-full w-10 mr-2">
-                    <UserCircleIcon className="h-9 text-fgd-3 w-9" />
-                  </div>
-                }
-              />
-            </div>
-          ) : (
-            <div className="pr-2 pl-2">
-              <img src={provider?.icon} className="h-5 w-5" />
-            </div>
-          )}
+        <div className="flex font-bold items-center text-fgd-1 text-left text-sm">
+          <div className="pr-2">
+            <img src={provider?.icon} className="h-5 w-5" />
+          </div>
+
           <div>
-            {connected && current?.publicKey ? (
-              <>
-                <DisplayAddress
-                  connection={connection.current}
-                  address={current?.publicKey}
-                  width="100px"
-                  height="20px"
-                  dark={true}
-                />
-                <StyledWalletProviderLabel className="font-normal text-fgd-3">
-                  {walletAddressFormatted}
-                </StyledWalletProviderLabel>
-              </>
-            ) : (
-              <>
-                Connect
-                <StyledWalletProviderLabel className="font-normal text-fgd-3">
-                  {provider?.name}
-                </StyledWalletProviderLabel>
-              </>
-            )}
+            {connected ? 'Disconnect' : 'Connect'}
+            <StyledWalletProviderLabel className="font-normal text-fgd-3">
+              {provider?.name}
```

Comment:
Maybe add

```
  if (!provider) {
    return null;
  }
```

at the top so you don't have to handle provider not being there `provider?.name` everywhere